### PR TITLE
Enable using winsqlite3.dll

### LINKED ISSUE #179: Enable using Windows in-the-box version of SQLite
Windows is shipping a version of SQLite in the box. We should look into enabling our library to use it.

### PR DESCRIPTION
**Pull request by bricelam**
To use, call `SqliteEngine.UseWinSqlite3()` before using SQLite.

Fixes #179

### CODE REVIEW

File: src/Microsoft.Data.Sqlite/Interop/VersionedMethods.cs [hunk 1]
```diff
@@ -7,67 +7,56 @@ namespace Microsoft.Data.Sqlite.Interop
...
-            var message = db == null || db.IsInvalid
-                ? _strategy.ErrorString(rc)
-                : NativeMethods.sqlite3_errmsg(db);
+        private static readonly BaseStrategy _strategy;
 
-            return Strings.FormatSqliteNativeError(rc, message);
+        static VersionedMethods()
+        {
+            var version = new Version(NativeMethods.sqlite3_libversion());
+            _strategy = version >= new Version(3, 7, 15)
```
Comment (natemcmaster) [inline src/Microsoft.Data.Sqlite/Interop/VersionedMethods.cs]: IMO the `if` statements are more readable than multi-nested ternary operators...but this is not a blocking objection.

File: src/Microsoft.Data.Sqlite/SqliteEngine.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Data.Sqlite.Interop;
+
+namespace Microsoft.Data.Sqlite
+{
+    public static class SqliteEngine
```
Comment (natemcmaster) [inline src/Microsoft.Data.Sqlite/SqliteEngine.cs]: Could we also add "UseDefault" to revert the engine back to `"sqlite3"`?

File: src/Microsoft.Data.Sqlite/Interop/VersionedMethods.cs (hunk 1 above)
Comment (bricelam) [inline src/Microsoft.Data.Sqlite/Interop/VersionedMethods.cs]: Will revert. I got a little carried away with renames and simplifications. The if statements did look better.

**Comment by natemcmaster** [comment]
:shipit:

File: src/Microsoft.Data.Sqlite/SqliteEngine.cs (hunk 2 above)
Comment (bricelam) [inline src/Microsoft.Data.Sqlite/SqliteEngine.cs]: I'll bring it up in a design meeting. It's an awkward API all around.

Comment (natemcmaster) [inline src/Microsoft.Data.Sqlite/SqliteEngine.cs]: Could this be a connection string parameter instead?

Comment (bricelam) [inline src/Microsoft.Data.Sqlite/SqliteEngine.cs]: Also note, you can't change it after it's loaded. If you don't want to use it, don't call it. Having another API to un-call it is silly.

Comment (bricelam) [inline src/Microsoft.Data.Sqlite/SqliteEngine.cs]: > Could this be a connection string parameter instead?

No, it's not per-connection. It's "global" (AppDomain) for the entire library.

Comment (natemcmaster) [inline src/Microsoft.Data.Sqlite/SqliteEngine.cs]: > No, it's not per-connection

...currently. But can't we dynamically switch to a different `ISqlite3` implementation?

Connection string seems like natural place to put this kind of configuration. Ideally the setting (something like`UseWinsqlite3=True;`) could handle the "if windows 10 do this, else do that"  for users so we wouldn't need different connection strings for Linux, Windows 7 and Windows 10.

Comment (bricelam) [inline src/Microsoft.Data.Sqlite/SqliteEngine.cs]: Could you file a bug? I agree it could be useful.

**Pull request merged**